public/js/NeighbourPlot.js

```diff
@@ -0,0 +1,168 @@
+                "<p><strong>Gene: </strong>" + d.gene_name + "</span></p>";
+        });
+
+    // setup canvas
+    var svg = d3.select('#neighbourpca').append('svg')
+        .attr('width', self.fullWidth)
+        .attr('height', self.fullHeight)
+        .append('g')
+        .attr('transform', 'translate(' + self.margin.left + ',' + self.margin.top + ')')
+        .call(tip);
+
+    // add points
+    svg.selectAll('circle')
+        .data(self.sampleData)
+        .enter()
+        .append('circle')
+        .attr('cx', function(d) {
+            return xScale(d.pca[0]);
+        })
+        .attr('cy', function(d) {
+            return yScale(d.pca[1]);
+        })
+        .attr('r', 5)
+        .classed('scatterpt', true)
+        .classed('activept', false)
+        .classed('neighbourpt', false)
+        .attr('id', function(d) {
+            return d._id;
+        })
+        .on('mouseover', tip.show)
+        .on('mouseout', tip.hide)
+        .on('click', function(d, i) {
+            var selection = d3.select(this);
+            if(!selection.classed('activept')) {
+                self.updatePoint(selection, d, i);
+            }
+        });
+
+    // append axis
+    svg.append('g')
+        .attr('class', 'x axis')
+        .attr('transform', 'translate(0,' + self.height + ')')
+        .call(xAxis);
+
+    svg.append('g')
+        .attr('class', 'y axis')
+        .call(yAxis);
+
+    // append axis labels
+    svg.append('text')
+        .attr('transform', 'translate(' + (self.width / 2) + ' ,' + (self.height + self.margin.bottom) + ')')
+        .style('text-anchor', 'middle')
+        .text('PC1');
+
+    svg.append('text')
+        .attr('transform', 'rotate(-90)')
+        .attr('y', 0 - self.margin.left)
+        .attr('x', 0 - self.height/2)
+        .attr('dy', '1em')
+        .style('text-anchor', 'middle')
+        .text('PC2');
+
+    // default select first point
+    var firstPoint = d3.select('.scatterpt');
+    console.log(firstPoint.data());
+    self.updatePoint(firstPoint, firstPoint.data()[0],  0);
+};
+
+NeighbourPlot.prototype.updatePlot = function() {
+
+};
+
+NeighbourPlot.prototype.updatePoint = function(selection, d, i) {
+    var self = this;
+
+    // update neighbour images
+    self.neighbourImages.getImages(d._id);
+    self.lineplot.drawLineplot(i);
+
+    d3.selectAll('.activept')
+        .classed('activept', false)
+        .classed('neighbourpt', false)
+        .transition()
+        .duration(125)
+        .attr('r', 5);
+
+    d3.selectAll('.neighbourpt')
+        .classed('neighbourpt', false)
+        .attr('r', 5)
+        .transition()
+        .duration(125);
+
+    // add neighbour class to all neighbours
+    var neighbours = d.neighbours;
+    for(var i = 1 ; i < neighbours.length; i++) {
+        d3.select('[id=' + neighbours[i] + ']')
+            .classed('neighbourpt', true)
+            .transition()
+            .duration(125)
```

Comment:
Here's a number ripe for codifying: you've typed "125" a total of... Well, not sure, but it feels like 125 times! =P A 'transition_duration' variable that can be set in one place would be really handy here...